File: Java/CodesInterviews/src/main/java/org/example/microsoft/ContainerWithMostWater.java
package org.example.microsoft;

public class ContainerWithMostWater {

    public int maxArea(int [] height) {
        int maxArea = 0;
        int l = 0;
        int r = height.length - 1;

        while (l < r) {
            maxArea = Math.max(maxArea, Math.min(height[l], height[r])* (r - l));

            if (height[l] < height[r]) {
                l += 1;
            } else {
                r -=1;
            }
        }
        return maxArea;
    }

    public static void main(String[] args) {
        int [] height = {1,8,6,2,5,4,8,3,7};
        ContainerWithMostWater containerWithMostWater = new ContainerWithMostWater();
        System.out.println(containerWithMostWater.maxArea(height));
    }
}

File: Kotlin/CodesInterviews/src/main/kotlin/microsoft/ContainerWithMostWater.kt
package microsoft

import kotlin.math.max
import kotlin.math.min

class ContainerWithMostWater {

    fun maxArea(height: IntArray): Int {
        var maxArea = 0
        var l = 0
        var r = height.size - 1

        while (l < r) {
            maxArea = max(maxArea, min(height[l], height[r]) * (r - l))

            if (height[l] < height[r]) {
                l += 1
            } else {
                r -= 1
            }
        }
        return maxArea
    }


}

fun main() {

    val height = intArrayOf(1,8,6,2,5,4,8,3,7)
    val water = ContainerWithMostWater()
    println(water.maxArea(height))

}
